compiler/src/iree/compiler/Dialect/Flow/Conversion/MeshToFlow/MeshToFlow.cpp

```diff
@@ -0,0 +1,789 @@
+  assert(mesh);
+  Value meshChannel = getDefaultChannel(mesh, useNamedDefaultChannels, builder);
+  SmallVector<Value> meshProcessMultiIndex =
+      builder.create<mesh::ProcessMultiIndexOp>(mesh).getResults();
+  SmallVector<Value> meshShape =
+      builder.create<mesh::ClusterShapeOp>(mesh).getResults();
+  SmallVector<Value> reorderedMeshIndex =
+      permute(ArrayRef<Value>(meshProcessMultiIndex), meshAxes);
+  SmallVector<Value> reorderedMeshShape =
+      permute(ArrayRef<Value>(meshShape), meshAxes);
+  SmallVector<Value> groupIndex =
+      filterOutByIndex(meshProcessMultiIndex, meshAxes);
+  SmallVector<Value> groupsShape = filterOutByIndex(meshShape, meshAxes);
+  Value reorderedProcessLinearIndex =
+      linearIndex(reorderedMeshIndex, reorderedMeshShape, builder);
+  Value color = linearIndex(groupIndex, groupsShape, builder);
+  return builder.create<ChannelSplitOp>(meshChannel, color,
+                                        reorderedProcessLinearIndex);
+}
+
+static SmallString<64> getChannelName(mesh::ClusterOp mesh,
+                                      ArrayRef<mesh::MeshAxis> axes) {
+  SmallString<64> res;
+  llvm::raw_svector_ostream stream(res);
+  stream << "_mesh_" << mesh.getSymName();
+  if (axes.empty()) {
+    return res;
+  }
+
+  stream << "_axes";
+  for (mesh::MeshAxis axis : axes) {
+    stream << "_" << axis;
+  }
+
+  return res;
+}
+
+static void buildChannelInitializer(mesh::ClusterOp mesh,
+                                    ArrayRef<mesh::MeshAxis> meshAxes,
+                                    bool useNamedDefaultChannels,
+                                    ImplicitLocOpBuilder &builder) {
+  Util::InitializerOp initOp = builder.create<Util::InitializerOp>();
+  Block *block = builder.createBlock(&initOp.getBody());
+  ImplicitLocOpBuilder::InsertionGuard insertionGuard(builder);
+  builder.setInsertionPointToStart(block);
+  Value channel =
+      buildChannelCreation(mesh, meshAxes, useNamedDefaultChannels, builder);
+  builder.create<Util::GlobalStoreOp>(channel, getChannelName(mesh, meshAxes));
+  builder.create<Util::InitializerReturnOp>();
+}
+
+// Construct a Flow channel inside `module` using
+// util.global and util.initializer.
+static void buildGlobalChannelCreation(mesh::ClusterOp mesh,
+                                       ArrayRef<mesh::MeshAxis> meshAxes,
+                                       bool useNamedDefaultChannels,
+                                       ModuleOp module, OpBuilder &opBuilder) {
+  if (isDefaultChannel(mesh, meshAxes)) {
+    return;
+  }
+
+  ImplicitLocOpBuilder builder(mesh.getLoc(), opBuilder);
+  ImplicitLocOpBuilder::InsertionGuard insertionGuard(builder);
+  builder.setInsertionPointToStart(&module.getBodyRegion().getBlocks().front());
+
+  auto channelName = getChannelName(mesh, meshAxes);
+  builder.create<Util::GlobalOp>(channelName, false,
+                                 Flow::ChannelType::get(builder.getContext()));
```

Comment:
style: use full IREE namespace here and elsewhere in the file (this protects us from others defining dialects/namespaces with the same prefix)
```suggestion
                                 IREE::Flow::ChannelType::get(builder.getContext()));
```

Reply:
I added the `IREE::` prefix in all places.